### ISSUE TITLE
Parsing error thrown for dynamic import in eslint v6.4.0

### ISSUE THREAD

**Issue by tituswoo** [eslint/eslint#12310]
**Tell us about your environment**

* **ESLint Version:** 6.4.0
* **Node Version:** v8.15.1
* **npm Version:** 5.10.0

**What parser (default, Babel-ESLint, etc.) are you using?**  default

**Please show your full configuration:**

<details>
<summary>Configuration</summary>

```json
{
    "plugins": [
        "react",
        "react-hooks"
    ],
    "extends": ["eslint:recommended", "plugin:react/recommended", "plugin:prettier/recommended"],
    "parserOptions": {
        "sourceType": "module",
        "ecmaVersion": 2020
    },
    "settings": {
        "react": {
            "version": "detect"
        }   
    },
    "rules": {
        "react/prop-types": "warn"
    },
    "env": {
        "browser": true,
        "es6": true
    }
}

```

</details>

**What did you do? Please include the actual source code causing the issue, as well as the command that you used to run ESLint.**

```js
import(`../blocks/${metadata.type}`).then(component => {
  setComponent(component)
})
```

```bash
eslint --quiet --fix src
```

**What did you expect to happen?**

Based on the [v6.2.0 release notes](https://eslint.org/blog/2019/08/eslint-v6.2.0-released#enhancements), I expected ESLint to not complain about the dynamic import.

**What actually happened? Please include the actual, raw output from ESLint.**

ESLint threw a parser error complaining about the dynamic import:

```
  22:5  error  Parsing error: 'import' and 'export' may only appear at the top level
✖ 1 problem (1 error, 0 warnings)
```

**Are you willing to submit a pull request to fix this bug?**

Yes

**Comment by mysticatea** [comment]
Thank you for this issue.

I couldn't reproduce it.

```
eslint test.js
Warning: React version was set to "detect" in eslint-plugin-react settings, but the "react" package is not installed. Assuming latest React version for linting.

C:\Users\t-nagashima.AD\dev\sandbox\test.js
  1:21  error  'metadata' is not defined      no-undef
  2:3   error  'setComponent' is not defined  no-undef

✖ 2 problems (2 errors, 0 warnings)

```

What the result of the following commands?

- `eslint --version`
- `npm ls -g espree` (there is `-g` because you look to use global-installed `eslint`.)

**Comment by tituswoo** [comment]
@mysticatea thank you so much for helping me debug this issue 😄 🔥 💯 

> (there is -g because you look to use global-installed eslint.)

👆 Your comment there gave me a light-bulb moment. I am using `lerna` for package management and it turns out that the way I had things set up meant the eslint CLI was calling my globally installed eslint instead of the eslint CLI I had installed in my lerna project. 😅 

So you were absolutely right, the eslint version I was _actually_ running was `eslint@v5.16.0` instead of what I _thought_ was running (`eslint@6.4.0`).

👏 Again, thank you so much for your help! I think we can mark this as closed now.
